libzeth/circuits/joinsplit.tcc

```diff
@@ -599,8 +600,8 @@ public:
         return get_inputs_bit_size() - (1 + NumOutputs) * FieldT::capacity();
     }
 
-    // Computes the number of field elements in the primary inputs
-    static size_t get_inputs_field_element_size()
+    // Computes the number of field elements in the public data
```

Comment:
"primary inputs/public input/instance" are the usual names used to denote the public assignment to the QAP variables, so - while i don't really mind the renamings in the file - I don't think this improves readability.

Reply:
I think th renaming here is quite important to reflect the fact that the joinsplit *gadget* does not control the *primary* inputs of the circuit.  I'm trying to consistently use `primary inputs` to be the actual primary inputs of the circuit, which may or may not coincide with the `public data` of the joinsplit gadget.  As part of making joinsplit more composable, the owner of the protoboard (not the joinsplit gadget itself) should be responsible for allocating and assigning `primary inputs` of the circuit, and the joinsplit gadget should be concerned only with its internal operations (which eventually assigns the public data elements).

Reply:
I see, thanks for the clarification @dtebbs - could you add a note (summarizing your comment above) and update the comments in the joinsplit source appropriately (i.e. `// --------- ALLOCATION OF PRIMARY INPUTS -------- //` etc etc)? That would be helpful.

Reply:
Latest commit has some updates to the joinsplit comments.  Let me know if anything is still unclear.

Reply:
Thanks